v2/tools/asoctl/cmd/log.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package cmd
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zerologr"
+	"github.com/rs/zerolog"
+	"github.com/vbauerster/mpb/v8"
+)
+
+var (
+	// Global logger and progress bar
+	// While these are global here, they can't be referenced from internal packages and should be passed through
+	// as parameters where needed.
+	log      logr.Logger
+	progress *mpb.Progress
```

Comment:
How does the progress bar work with no `--output` flag? Will it break piping?

Reply:
Having any logging at all will break piping, so it's not the progress bar that breaks it.

Maybe we need a --silent flag to suppress logging?

Reply:
This is why many Linux tools emit their logs to `stderr` rather than `stdout`, I think.

Structured output goes to `stdout`, log output goes to `stderr` always, and then piping works by default.

If piping doesn't work, then `-o` being an optional flag seems somewhat pointless, because there's not a lot of value in running the tool and emitting to `stdout` given you can't actually pipe that to a file (or to `kubectl` directly)?